sdk/identity/Azure.Identity/tests/AuthenticationRecordTests.cs

```diff
@@ -0,0 +1,104 @@
+            Assert.ThrowsAsync<ArgumentNullException>(async () => await AuthenticationRecord.DeserializeAsync(null));
+        }
+
+        [Test]
+        public async Task SerializeDeserializeAsync()
+        {
+            var expRecord = new AuthenticationRecord(Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString());
+
+            byte[] buff = new byte[256];
+
+            var stream = new MemoryStream(buff);
+
+            await expRecord.SerializeAsync(stream);
+
+            // decode to string and re-encode to strip extra bytes
+            var buffStr = Encoding.UTF8.GetString(buff).Trim('\0');
+
+            buff = Encoding.UTF8.GetBytes(buffStr);
+
+            stream = new MemoryStream(buff);
+
+            var actRecord = await AuthenticationRecord.DeserializeAsync(stream);
+
+            Assert.AreEqual(expRecord.Username, actRecord.Username);
+            Assert.AreEqual(expRecord.Authority, actRecord.Authority);
+            Assert.AreEqual(expRecord.HomeAccountId, actRecord.HomeAccountId);
+            Assert.AreEqual(expRecord.TenantId, actRecord.TenantId);
+        }
+
+        [Test]
+        public void SerializeDeserialize()
+        {
+            var expRecord = new AuthenticationRecord(Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString());
+
+            byte[] buff = new byte[256];
+
+            var stream = new MemoryStream(buff);
+
+            expRecord.Serialize(stream);
+
+            // decode to string and re-encode to strip extra bytes
+            var buffStr = Encoding.UTF8.GetString(buff).Trim('\0');
+
+            buff = Encoding.UTF8.GetBytes(buffStr);
+
+            stream = new MemoryStream(buff);
+
+            var actRecord = AuthenticationRecord.Deserialize(stream);
+
+            Assert.AreEqual(expRecord.Username, actRecord.Username);
+            Assert.AreEqual(expRecord.Authority, actRecord.Authority);
+            Assert.AreEqual(expRecord.HomeAccountId, actRecord.HomeAccountId);
+            Assert.AreEqual(expRecord.TenantId, actRecord.TenantId);
+        }
+
+        [Test]
+        public void SerializeCancellationTokenCancelled()
+        {
+            CancellationTokenSource cts = new CancellationTokenSource();
+
+            cts.Cancel();
+
+            var expRecord = new AuthenticationRecord(Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString(), Guid.NewGuid().ToString());
+
+            var stream = new MemoryStream(new byte[256]);
+
+            Assert.CatchAsync<OperationCanceledException>(async () => await expRecord.SerializeAsync(stream, cts.Token));
+        }
+
+        [Test]
+        public void DeserializeCancellationTokenCancelled()
+        {
+            CancellationTokenSource cts = new CancellationTokenSource();
+
+            cts.Cancel();
+
+            var stream = new MemoryStream(new byte[256]);
```

Comment:
you know MemoryStream has a ctor that takes an int?

Reply:
Now I do :)